skopt/space/space.py

```diff
@@ -69,10 +69,10 @@ def check_dimension(dimension, transform=None):
     if len(dimension) == 2:
         if any([isinstance(d, str) for d in dimension]):
             return Categorical(dimension, transform=transform)
-        elif any([isinstance(dim, float) for dim in dimension]):
```

Comment:
In the old version it checks for floats before integers. With the new it checks for integers first. If not, any intended `Integer` dimension will actually be transformed into a `Real` dimension.